### PR TITLE
Added Autocode show messages module

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Hi @as2388 please take a look pending this review https://github.com/AfricasVoices/Test- from @lukechurch .

### CODE REVIEW

File: docker-run.sh [hunk 1]
```diff
@@ -106,11 +106,11 @@ docker start -a -i "$container"
...
 
-# mkdir -p "$OUTPUT_ICR_DIR"
-# docker cp "$container:/data/output-icr/." "$OUTPUT_ICR_DIR"
+mkdir -p "$OUTPUT_ICR_DIR"
+docker cp "$container:/data/output-icr/." "$OUTPUT_ICR_DIR"
 
-# mkdir -p "$OUTPUT_CODED_DIR"
-# docker cp "$container:/data/output-messages-csv" "$OUTPUT_MESSAGES_CSV"
+mkdir -p "$OUTPUT_CODED_DIR"
+docker cp "$container:/data/coded/." "$OUTPUT_CODED_DIR"
```
Comment (as2388) [inline docker-run.sh]: Took me a while to work out whether this was before or after coding.
Suggest we both rename this to :/data/auto-coded-coda-datasets $OUTPUT_AUTO_CODED_CODA_DATASETS

File: project_test/auto_code_show_messages.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    TEST_KEYS = []
+    for plan in PipelineConfiguration.TEST_SHOWS_CODING_PLANS:
+        TEST_KEYS.append(plan.raw_field)
+
+    SENT_ON_KEY = "sent_on"
+    NOISE_KEY = "noise"
+    PROJECT_START_DATE = isoparse("2019-01-01T00:00:00+03:00")
+    PROJECT_END_DATE = isoparse("2019-04-01T00:00:00+03:00")
+    ICR_MESSAGES_COUNT = 1 # Since we have few test messages
+    ICR_SEED = 0
```
Comment (as2388) [inline project_test/auto_code_show_messages.py:26]: No need to do anything here, but just to note that these constants will soon be moving into pipeline configuration json

File: project_test/auto_code_show_messages.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    @classmethod
+    def auto_code_show_messages(cls, user, data, icr_output_dir, coda_output_dir):
```
Comment (as2388) [inline project_test/auto_code_show_messages.py:29]: Let me know if you changed anything in this function

File: project_test/lib/channels.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            return False 
+    
+    @classmethod
+    def set_channel_keys(cls, user, data, time_key):
+        for td in data:
+            timestamp = isoparse(td[time_key])
+
+            channel_dict = dict()
+            
+            # Set time as NON_LOGICAL if it doesn't fall in range of ** radio_show**
```
Comment (as2388) [inline project_test/lib/channels.py]:
```suggestion
            # Set time as NON_LOGICAL if it doesn't fall in range of the **radio_shows**
```

File: project_test/lib/channels.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    #TODO: Setup rapid pro shows to test Ad,promos,shows & date ranges then update the list below.
+    NON_LOGICAL_KEY = "non_logical_time"
+    INTERNET_KEY = "internet_working"
+    WATER_KEY =  "water_filter"
+    WASTE_KEY = "waste_disposal"
+
+    # Time ranges expressed in format (start_of_range_inclusive, end_of_range_exclusive)
+    INTERNET_RANGES = [("2019-01-01T00:00:00+03:00","2019-04-01T00:00:00+03:00")]
+    WATER_RANGES = [("2019-01-01T00:00:00+03:00","2019-04-01T00:00:00+03:00")]
+    WASTE_RANGES = [("2019-01-01T00:00:00+03:00","2019-04-01T00:00:00+03:00")]
```
Comment (as2388) [inline project_test/lib/channels.py:19]: Following my offline questions about flow configurations for this project, please add a comment explaining why all these time ranges are the same.

**Review by as2388 — APPROVED** [review]

File: project_test/lib/channels.py (hunk 5 above)
Comment (IsaackMwenda) [reply project_test/lib/channels.py:19]: Thanks @as2388.

**Pull request merged**